protocol/parser/parser.go

```diff
@@ -163,6 +237,118 @@ func parse(rpcInput RPCInput, blockParser spectypes.BlockParser, dataSource int)
 	return retval, nil
 }
 
+type ParsedInput struct {
+	parsedBlock  int64
+	parsedHashes []string
+}
+
+func NewParsedInput() *ParsedInput {
+	return &ParsedInput{
+		parsedBlock:  spectypes.NOT_APPLICABLE,
+		parsedHashes: make([]string, 0),
+	}
+}
+
+func (p *ParsedInput) SetBlock(block int64) {
+	p.parsedBlock = block
+}
+
+func (p *ParsedInput) GetBlock() int64 {
+	return p.parsedBlock
+}
+
+func (p *ParsedInput) GetBlockHashes() ([]string, error) {
+	if len(p.parsedHashes) == 0 {
+		return nil, fmt.Errorf("no parsed hashes found")
+	}
+	return p.parsedHashes, nil
+}
+
+func ParseWithGenericParsers(rpcInput RPCInput, genericParsers []spectypes.GenericParser) (*ParsedInput, error) {
+	if len(genericParsers) == 0 {
+		return nil, fmt.Errorf("no generic parsers to use")
+	}
+
+	params := rpcInput.GetParams()
+	if params == nil {
+		return nil, utils.LavaFormatTrace("rpcInput params are nil", utils.LogAttr("rpcInput", rpcInput))
+	}
+
+	// We try to parse the params with all the generic parsers, the first one that succeeds, its value is returned
+	for _, genericParser := range genericParsers {
+		retval, err := parseGeneric(params, genericParser)
+		if err == nil {
+			return retval, nil
+		}
+	}
+
+	// we didn't find a generic parser that worked
+	return nil, utils.LavaFormatTrace("failed to parse with generic parsers",
+		utils.LogAttr("params", rpcInput.GetParams()),
+		utils.LogAttr("jsonParams", params),
+		utils.LogAttr("genericParsers", genericParsers),
+	)
+}
+
+func parseGeneric(input interface{}, genericParser spectypes.GenericParser) (*ParsedInput, error) {
+	value, err := findGenericParserValue(input, genericParser)
+	if err != nil {
+		return nil, err
+	}
+
+	utils.LavaFormatTrace("parsed generic value",
+		utils.LogAttr("input", input),
+		utils.LogAttr("genericParser", genericParser),
+		utils.LogAttr("value", value),
+	)
+
+	switch genericParser.ParseType {
+	case spectypes.PARSER_TYPE_BLOCK_LATEST:
```

Comment:
block latest is going to be used instead of the current block parsing, so you can;t always ovveride latest when it appears, what you need to do is only override with latest if there's a value in the parser that is equal "latest"